### PR TITLE
[xh]Alogia integration

### PR DESCRIPTION
**Pull request by matrixstone**
# Description
Algolia is a hosted search engine can be used for context searching and vector database: https://www.algolia.com/doc/. It is requested by customer to support it.

# How Has This Been Tested?
Mage UI:

Loader:
test1
loader_result

Exporter:
exporter_template
data_written

# Checklist
- [X] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [X] I have performed a self-review of my own code
- [] I have added unit tests that prove my fix is effective or that my feature works
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] If new documentation has been added, relative paths have been added to the appropriate section of `docs/mint.json`

cc:
@dy46

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -48,6 +48,7 @@ typing-extensions==4.5.0
 
 # extras
 astor>=0.8.1
+algoliasearch>=3.0.0
```
Comment (dy46) [inline requirements.txt:51]: can we restrict the package version here as well

Comment (matrixstone) [reply requirements.txt]: Locked

**Comment by dy46** [comment]
same for this one, can you add the new dependencies to the documentation instead of the requirements.txt/setup.py files?

**Comment by matrixstone** [comment]
Also updated in this PR. I remove the library from `requirements.txt` and `setup.py`. Instead updated the documentation file. Thank you David! @dy46

File: docs/integrations/databases/Algolia.mdx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+default:
+  ALGOLIA_APP_ID: app_id
+  ALGOLIA_API_KEY: api_key
+  ALGOLIA_INDEX_NAME: index_name
+```
+
+## Required Algolia Libraries
+
+1. Add `algoliasearch==3.0.0 into `requirements.txt` to install the required weaviate library.
+2. Add `algoliasearch==3.0.0` into `setup.py` under the `all` section.
```
Comment (dy46) [inline docs/integrations/databases/Algolia.mdx:21]: The users can just add the package to their Mage `requirements.txt` file, no need to add it to tthe `setup.py` file

**Review by dy46 — APPROVED** [review]
left one more comment on the documentation

File: docs/integrations/databases/Algolia.mdx (hunk 2 above)
Comment (matrixstone) [reply docs/integrations/databases/Algolia.mdx]: Updated

**Pull request merged**